src/main/java/lotto/Rank.java

```diff
@@ -0,0 +1,49 @@
+package lotto;
+
+public enum Rank {
+    NONE(0, 0),
+    FIFTH(5000, 3),
+    FOURTH(50000, 4),
+    THIRD(1500000, 5),
+    SECOND(30000000, 5),
+    FIRST(2000000000, 6);
+
+    private final int winningAmount;
+    private final int matchCount;
+
+    Rank(final int winningAmount, final int matchCount) {
+        this.winningAmount = winningAmount;
+        this.matchCount = matchCount;
+    }
+
+    public int getMatchCount() {
+        return matchCount;
+    }
+
+    public int getWinningAmount() {
+        return winningAmount;
+    }
+
+    public static Rank checkRank(final int matchCount, final boolean hasBonusNumber) {
+        for (final Rank rank : Rank.values()) {
+            if (matchCount == SECOND.matchCount) {
+                return checkSecondOrThird(hasBonusNumber);
+            }
+            if (rank.matchCount == matchCount) {
+                return rank;
+            }
+        }
+        return NONE;
```

Comment:
아래와 같이 변경해볼 수도 있을 것 같네요. 🤗 
```suggestion
        if (matchCount == SECOND.matchCount) {
            return checkSecondOrThird(hasBonusNumber);
        }

        return Arrays.stream(Rank.values())
            .filter(rank -> rank.matchCount == matchCount)
            .findAny()
            .orElse(NONE);
```

Reply:
> (질문) 사실 로키가 지적해주신 부분에 대해 페어랑 많이 고민해보았었습니다.
처음 작성한 코드는 그렇지 않았는데,
프로그래밍 요구사항인 메서드 depth 문제를 해결하려다 보니 그런 코드로 변해간 거 같아요😢
우선 모든 요소에 대해 분기문을 가지지 않도록 로직을 수정했는데,
depth 문제(for문안에 if문)를 해결할 수 있는 좋은 방법이 있을까요?

상황에 따라 다르기 때문에 딱 잘라서 말하기는 어렵지만 
결국에 depth를 줄이기위한 가장 쉬운 접근법은 메서드를 추출하는 것이라 생각해요. 😃
답변이 제가 말하고도 조금 아쉬운 것 같은데... 혹시 depth 줄이기에 고민되는 상황이 발생하면 공유주세요. 😁